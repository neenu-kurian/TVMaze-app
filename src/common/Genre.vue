<template>
  <div v-if="genre" class="show__genre--wrapper">
    <span class="show__genre">{{genre}}</span>
  </div>
</template>

<script>
export default {
  name: 'Genre',
  props: ['genre']
}
</script>
