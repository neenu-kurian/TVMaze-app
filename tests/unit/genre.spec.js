import { mount } from '@vue/test-utils'
import Genre from '@/common/Genre'

describe('Genre', () => {
  it('displays genre correctly', () => {
    const wrapper = mount(Genre, {
      propsData: {
        genre: 'comedy'
      }
    })

    expect(wrapper.text()).toContain('comedy')
  })

  it('renders the component correctly', () => {
    const wrapper = mount(Genre, {
      propsData: {
        genre: 'drama'
      }
    })

    expect(wrapper).toMatchSnapshot()
  })

  it('doesnt display genre if it not available', () => {
    const wrapper = mount(Genre)

    expect(wrapper.find('.show__genre--wrapper').exists()).toBe(false)
  })
})
