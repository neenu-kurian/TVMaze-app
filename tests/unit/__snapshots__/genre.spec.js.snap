// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Genre renders the component correctly 1`] = `<div class="show__genre--wrapper"><span class="show__genre">drama</span></div>`;

exports[`renders the component correctly 1`] = `<div class="show__genre--wrapper"><span class="show__genre">drama</span></div>`;
